myuw/static/js/test/global.js

```diff
@@ -0,0 +1,119 @@
+var path = require("path");
+var fs = require("fs");
+var sinon = require("sinon");
+
+var Environment = {
+    _stub: null,
+
+    init: function (config) {
+        if (!config) {
+            config = {};
+        }
+
+        // create test document
+        var window = require('jsdom').jsdom().defaultView;
+
+        // pull in supporting tools
+        var assert = require("assert");
+
+        // pull in common tools
+        var $ = require('jquery')(window);
+        var Handlebars = require("../../vendor/js/handlebars-v2.0.0.js");
+        var Moment = require("../../vendor/js/moment.2.8.3.min.js");
+
+        // setup common client tools
+        global.$ = $;
+        global.window = window;
+        global.assert = assert;
+        global.Handlebars = Handlebars;
+        global.moment = Moment;
+        var HandlebarsHelpers = require("../handlebars-helpers.js")
+
+        // set up client environment
+        window.user = {};
+        window.user.student = true;
+
+        // default test term
+        window.term = {};
+        window.term.year = 2013;
+        window.term.quarter = 'spring';
+
+        if (config.hasOwnProperty('render_id')) {
+            $('body').append($('<div/>', { 'id': config.render_id }));
+        }
+
+        // stub onload init routines
+        window.RenderPage = function () {};
+        global.Profile = { add_events: function () {} };
+        global.Modal = { add_events: function () {} };
+        global.LogUtils = {
+            init_logging: function () {},
+            cardLoaded: function(name, el) {
+                $(window).trigger("myuw:card_load", [ name, el ])
+            }
+        };
+
+        // pull in scripts
+        Environment._load_script('../myuw_m.js');
+        Environment._load_script('../ws_data.js');
+        if (config.hasOwnProperty('scripts')) {
+            $.each(config.scripts, function () {
+                Environment._load_script(this.toString());
+            });
+        }
+
+        // pull in templates
+        if (config.hasOwnProperty('templates')) {
+            $.each(config.templates, function () {
+                Environment._load_template(this.toString());
+            });
+        }
+    },
+    ajax_stub: function (json_data_file) {
+        var json_file = path.join(__dirname, 'ajax', json_data_file);
+        var json_data = JSON.parse(fs.readFileSync(json_file));
+        Environment._stub = sinon.stub($, 'ajax');
+        Environment._stub.yieldsTo('success', json_data);
+    },
+    ajax_stub_restore: function () {
+        if (Environment._stub) {
+            Environment._stub.restore();
+        }
+    },
+    _load_script(script) {
```

Comment:
What do you think about adding a check here to see if there's already a matching name in global?  It probably wouldn't come up often, but if the test suite could say something like 'Unable to load CardFoo from js/card/not_foo.js, already loaded from js/card/foo.js' it might save someone some grief